fe/fe-core/src/main/java/org/apache/doris/load/GroupCommitManager.java

```diff
@@ -0,0 +1,171 @@
+                    }
+                    continue;
+                }
+                TStatusCode code = TStatusCode.findByValue(response.getStatus().getStatusCode());
+                if (code != TStatusCode.OK) {
+                    String msg = "get wal queue size fail,backend id: " + backend.getId() + ", status: "
+                            + response.getStatus();
+                    LOG.warn(msg);
+                    try {
+                        Thread.sleep(100);
+                    } catch (InterruptedException ie) {
+                        LOG.info("group commit manager sleep wait InterruptedException: ", ie);
+                    }
+                    continue;
+                }
+                size = response.getSize();
+                done = true;
+            }
+            if (size > 0) {
+                LOG.info("backend id:" + backend.getId() + ",wal size:" + size);
+                empty = false;
+            }
+        }
+        return empty;
+    }
+
+    public long getAllWalQueueSize(Backend backend) {
+        PGetWalQueueSizeRequest request = PGetWalQueueSizeRequest.newBuilder()
+                .build();
+        PGetWalQueueSizeResponse response = null;
+        long start = System.currentTimeMillis();
+        boolean done = false;
+        long size = 0;
+        while (!done && System.currentTimeMillis() - start <= Config.check_wal_queue_timeout_threshold) {
+            try {
+                Future<PGetWalQueueSizeResponse> future = BackendServiceProxy.getInstance()
+                        .getAllWalQueueSize(new TNetworkAddress(backend.getHost(), backend.getBrpcPort()), request);
+                response = future.get();
+            } catch (Exception e) {
+                LOG.warn("encounter exception while getting all wal queue size on backend id: " + backend.getId()
+                        + ",exception:" + e);
+                try {
+                    Thread.sleep(100);
+                } catch (InterruptedException ie) {
+                    LOG.info("group commit manager sleep wait InterruptedException: ", ie);
+                }
+                continue;
+            }
+            TStatusCode code = TStatusCode.findByValue(response.getStatus().getStatusCode());
+            if (code != TStatusCode.OK) {
+                String msg = "get all wal queue size fail,backend id: " + backend.getId() + ", status: "
+                        + response.getStatus();
+                LOG.warn(msg);
+                try {
+                    Thread.sleep(100);
+                } catch (InterruptedException ie) {
+                    LOG.info("group commit manager sleep wait InterruptedException: ", ie);
+                }
+                continue;
+            }
+            size = response.getSize();
+            done = true;
+        }
+        if (size > 0) {
+            LOG.info("backend id:" + backend.getId() + ",all wal size:" + size);
+        }
+        return size;
+    }
+
```

Comment:
Consider the use of Optional<TransactionState> to avoid returning null and provide a more functional approach.